omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,80 @@
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import static org.openmrs.ConditionVerificationStatus.CONFIRMED;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Encounter;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_2;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Tests functionality of {@link DiagnosisController2_2}.
+ */
+public class DiagnosisController2_2Test extends MainResourceControllerTest {
+
+	private DiagnosisService diagnosisService;
+	private Encounter encounter;
+
+	@Before
+	public void before() throws Exception {
+		this.diagnosisService = Context.getDiagnosisService();
+		this.encounter = new Encounter();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return RestConstants.VERSION_1 + "/patientdiagnoses";
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.DIAGNOSIS_UUID;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return diagnosisService.getPrimaryDiagnoses(this.encounter).size();
+	}
+
+	@Test
+	public void shouldCreateADiagnosis() throws Exception {
```

Comment:
Can you add a few more tests? Like for fetching existing diagnoses, voiding, etc